javascript/javascript1/week4/Voice assistant/voice-assistant.js

```diff
@@ -1,28 +1,122 @@
+"use strict";
 
-const userInformation = {
-    name: '',
-    toDoList: [],
-    day: '',
-    favoriteDish: '',
-    myCalendar: []
-};
+let names = [];
+let whoAmI = 'What is my name';
+let day = 'What day is it today';
+let todo = [];
+let myTodoList = 'What is on my todo';
+let favoriteDish = [];
+let event = [];
+let eventList = 'What am I doing this week';
 
-  //Introducing myself
-  function getReply(command) {
-    if(command.toString().toLowerCase() === 'hello my name is Benjamin'.toLowerCase()){
-    return ('nice to meet you Benjamin');
-    } else if(command.toString().toLowerCase() === 'what is my name'.toLowerCase()){
-        if(userInformation.name === undefined){
-            return `You have not introduced yourself`;
-        } else {
-        return `${userInformation.name}`;
-     } 
+function getReply(command) {
+
+    // 1. To save the person's name and returning nice to meet you along the person's name
+    if (command.startsWith('Hello') && command.includes('name')) {
+        let commandArray = command.split(' ')
+        let name = commandArray[commandArray.length - 1]
+        names.push(name)
+        return `Nice to meet you ${names[names.length -1]}`;
     }
 
+    //If you ask for your name and you have introduced yourself you'll get your name
+    if (command === whoAmI && names.length !== 0) {
+        return `Your  ${names[names.length -1]}`;
+    }
+    //If you ask for your name without introducing yourself
+    if (command === whoAmI && names.length === 0) {
+        return `Please tell me your name and I'll remember it for next time`
+    }
+    //Logging out what day today is in format 02. April 2020
+    if (command === day) {
+        let monthsOfTheYear = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+        const today = new Date();
+        const day = today.getDate();
+        const month = today.getMonth();
+        const year = today.getFullYear();
+        return `Today is ${day}. of ${monthsOfTheYear[month]} ${year}`;
+    }
+    if (command.includes('what is') && command.split(' ')[command.split(' ').length - 1] === Number) {
+        let mathEquation = command.split(' ')
+        console.log(mathEquation)
+    }
+    //Adding things to todo
+    if (command.startsWith("Add") && command.endsWith("todo")) {
+        let addTodoWord = command.split(' ');
+        let wordToAdd = addTodoWord[1]
+        todo.push(wordToAdd);
+        return `${wordToAdd} added to your todo`;
+    }
+    //Removing things from todo
+    if (command.startsWith('Remove') && command.endsWith('todo')) {
+        let removeTodoWord = command.split(' ');
+        let wordToRemove = removeTodoWord[1]
```

Comment:
you should update `todo` object here. What do you thing? Maybe think of a method that removes a string from the `todo` simillar to adding a string to `todo`